target/linux/ramips/patches-5.4/201-4-mips-ralink-add-zboot-debug-uart-code.patch

```diff
@@ -0,0 +1,49 @@
+From 91cb61dd7786c17fa0e804b8ba82ee2dd83091a6 Mon Sep 17 00:00:00 2001
+From: John Thomson <git@johnthomson.fastmail.com.au>
+Date: Thu, 15 Oct 2020 14:23:25 +1000
+Subject: [PATCH 4/4] mips: ralink: add zboot debug uart code
+
+Signed-off-by: John Thomson <git@johnthomson.fastmail.com.au>
+---
+ arch/mips/Kconfig                  | 1 +
+ arch/mips/boot/compressed/Makefile | 5 +++++
+ 2 files changed, 6 insertions(+)
+
+diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
+index 5f7c0c5d01e..033308ab3f3 100644
+--- a/arch/mips/Kconfig
++++ b/arch/mips/Kconfig
+@@ -625,6 +625,7 @@ config RALINK
+ 	select SYS_SUPPORTS_LITTLE_ENDIAN
+ 	select SYS_SUPPORTS_MIPS16
+ 	select SYS_SUPPORTS_ZBOOT
++	select SYS_SUPPORTS_ZBOOT_UART_PROM
+ 	select SYS_HAS_EARLY_PRINTK
+ 	select CLKDEV_LOOKUP
+ 	select ARCH_HAS_RESET_CONTROLLER
+diff --git a/arch/mips/boot/compressed/Makefile b/arch/mips/boot/compressed/Makefile
+index 8314e800b25..e88bafc07a6 100644
+--- a/arch/mips/boot/compressed/Makefile
++++ b/arch/mips/boot/compressed/Makefile
+@@ -39,6 +39,7 @@ ifdef CONFIG_DEBUG_ZBOOT
+ vmlinuzobjs-$(CONFIG_DEBUG_ZBOOT)		   += $(obj)/dbg.o
+ vmlinuzobjs-$(CONFIG_SYS_SUPPORTS_ZBOOT_UART16550) += $(obj)/uart-16550.o
+ vmlinuzobjs-$(CONFIG_SYS_SUPPORTS_ZBOOT_UART_PROM) += $(obj)/uart-prom.o
++vmlinuzobjs-$(CONFIG_RALINK)			   += $(obj)/uart-prom-ralink.o
+ vmlinuzobjs-$(CONFIG_MIPS_ALCHEMY)		   += $(obj)/uart-alchemy.o
+ vmlinuzobjs-$(CONFIG_ATH79)			   += $(obj)/uart-ath79.o
+ endif
+@@ -47,6 +48,10 @@ extra-y += uart-ath79.c
+ $(obj)/uart-ath79.c: $(srctree)/arch/mips/ath79/early_printk.c
+ 	$(call cmd,shipped)
+ 
++extra-y += uart-prom-ralink.c
++$(obj)/uart-prom-ralink.c: $(srctree)/arch/mips/ralink/early_printk.c
```

Comment:
I'd argue that the existence of early_printk.c for mt7621/mt7628 is a mistake, as these two SoCs have 16550 compatible uart controller. It should reuse existing kernel code for early printk instead.
Anyway this patch is a bit off-topic for this PR. Maybe just drop it?

Reply:
I would _like_ to leave the facility to debug the decompressor, but leave `#CONFIG_DEBUG_ZBOOT is not set`. While the output is minimal, it may help troubleshoot:
```
zimage at:     80932420 80D61FDE
Uncompressing Linux at load address 80001000
Now, booting the kernel...
[    0.000000] Linux version 5.4.70 (john@media01) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r0+13048-e8ead3f888)) #0 SMP Thu Oct 15 04:47:15 2020
```

Fair enough, I just picked including the ralink [`prom`/`early_printk`](https://elixir.bootlin.com/linux/v5.4.70/source/arch/mips/ralink/early_printk.c) code path because I did not want to set values that were already somewhere in kernel.

If you think the 16550 code path is more appropriate, can I include [your patch](https://github.com/openwrt/openwrt/pull/3511#issuecomment-708491373), or would you prefer we left debug out?

Cheers

Reply:
I've sent that patch upstream: https://lkml.org/lkml/2020/10/15/34
so I think we could leave it out for now and backport it when it's applied upstream.